rholang/src/test/scala/coop/rchain/rholang/interpreter/CostAccountingReducerTest.scala

```diff
@@ -65,28 +66,70 @@ class CostAccountingReducerTest extends FlatSpec with Matchers with TripleEquals
   }
 
   it should "stop if OutOfPhloError is returned from RSpace" in {
-    val tuplespaceAlg = new Tuplespace[Task] {
-      override def produce(
-          chan: Par,
-          data: ListParWithRandom,
-          persistent: Boolean,
-          sequenceNumber: Int
-      ): Task[Unit] =
-        Task.raiseError(OutOfPhlogistonsError)
-      override def consume(
-          binds: Seq[(BindPattern, Par)],
-          body: ParWithRandom,
-          persistent: Boolean,
-          sequenceNumber: Int
-      ): Task[Unit] = Task.raiseError(OutOfPhlogistonsError)
-    }
 
-    implicit val errorLog = new ErrorLog[Task]()
-    implicit val rand     = Blake2b512Random(128)
-    implicit val cost     = CostAccounting.initialCost[Task](Cost(1000)).runSyncUnsafe(1.second)
-    val reducer           = new DebruijnInterpreter[Task, Task.Par](tuplespaceAlg, Map.empty)
-    val send              = Send(Par(exprs = Seq(GString("x"))), Seq(Par()))
-    val test              = reducer.inj(send).attempt.runSyncUnsafe(1.second)
+    val iSpace =
+      new ISpace[Task, Par, BindPattern, ListParWithRandom, ListParWithRandom, TaggedContinuation] {
```

Comment:
Holy moly! Could this have been the reason for having the Tuplespace trait? :thinking:

Please extract an `ISpaceStub` class that will leave all methods undefined, and allow for the test to be as readable as previously.

Reply:
What do you mean extract? In a separate file? At the top of the test? You basically want me to move it somewhere else and keep the implementation?

Reply:
I want you to provide a:

```scala
class ISpaceStub extends ISpace {
  //all methods implemented using ??? - it's ok for a stub
}
```

so that the test can reduce its setup to what +/- it was previously
```scala
   val ispace = new ISPaceStub {
      override def produce(
          chan: Par,
          data: ListParWithRandom,
          persistent: Boolean,
          sequenceNumber: Int
      ): Task[Unit] =
        Task.raiseError(OutOfPhlogistonsError)
      override def consume(
          binds: Seq[(BindPattern, Par)],
          body: ParWithRandom,
          persistent: Boolean,
          sequenceNumber: Int
      ): Task[Unit] = Task.raiseError(OutOfPhlogistonsError)
    }
```

At a minimum, put the ISpaceStub at the bottom of the file, or in a separate file if you will.